dotnet/src/Microsoft.AutoGen/AgentChat/Abstractions/Messages.cs

```diff
@@ -22,6 +22,11 @@ public abstract class AgentMessage
     /// </summary>
     public required string Source { get; set; }
 
+    /// <summary>
+    /// The <see cref="IChatClient"/> usage incurred when producing this message.
+    /// </summary>
+    public RequestUsage? ModelUsage { get; set; }
```

Comment:
Perhaps we can add this in the future. Currently in Python agentchat, we are still deciding how to best surfacing the model usage. And relying on the model usage field for termination may not be a great idea. It think it's better to monitor the model client usage in a separate coroutine and trigger an external termination.